drivers/net/bnxt/bnxt_ethdev.c

```diff
@@ -963,6 +963,45 @@ static int bnxt_flow_ctrl_set_op(struct rte_eth_dev *dev,
 	return bnxt_set_hwrm_link_config(bp, true);
 }
 
+static int bnxt_set_vf_rate_limit(struct rte_eth_dev *eth_dev, uint16_t vf,
+				uint16_t tx_rate, uint64_t q_msk)
+{
+	struct bnxt *bp = (struct bnxt *)eth_dev->data->dev_private;
+	uint16_t tot_rate = 0;
+	uint64_t idx;
+	int rc;
+
+	if (!bp->pf.active_vfs)
+		return -EINVAL;
+
+	if (vf >= BNXT_MAX_VFS(bp))
+		return -EINVAL;
+
+	/* Add up the per queue BW and configure MAX BW of the VF */
+	for (idx = 0; idx < 64; idx++) {
+		if ((1ULL << idx) & q_msk)
+			tot_rate += tx_rate;
+	}
```

Comment:
That is right. This was the discussion I had initiated few months back with Hemal. You were on CC. :-) 

DPDK API uses a per queue VF QOS. While HWRM does not have that.
So till we get that, I suggested arriving at a sum of per queue Tx rate.
I did not hear anything negative on that. So I went ahead.

Do you want to hold this?